units/src/amount/result.rs

```diff
@@ -182,6 +182,11 @@ crate::internal_macros::impl_op_for_references! {
 
         fn div(self, rhs: u64) -> Self::Output { self.and_then(|lhs| lhs / rhs) }
     }
+    impl ops::Div<Amount> for Amount {
+        type Output = u64;
```

Comment:
Oh, wait, why is this not `NumOpResult<u64>`? The RHS may be zero.

Also I think your reason for not having `Div<NumOpResult<Amount>>` somehow assumes it's impossible to return `NumOpResult` but why would it be the case?

Reply:
AFAIU the point of the `NumOpResult` is to protect against overflow bugs not to prevent panics. `Div` panics if RHS is zero for stdlib types and here - I don't see a problem with that?

Also I just have a gut feeling that we should only use the two amount types as `T` in `NumOpResult<T>` but I can't quite put my finger on why.

Reply:
Let's move "should div by 0 return a result or panic" to a separate issue.

>I'm really not sure about the Output thing. 

What `Output` thing?

Reply:
This one right here I think he means, line 186.

Reply:
@tcharding you're not wrong and also the consequences might be less severe than with overflow. But isn't it quite weird that all operations return `NumOpResult` except for `Div`? I'm worried it'd get confusing.

Regarding your gut feeling, I have a good reason to at least use `()` in there, so I don't see why `u64` would be bad. We would really want to use `Result` which already allows anything but can't because of orphan rules.

Reply:
I'm not convinced, please raise an issue if you are concerned. Otherwise due to your comment https://github.com/rust-bitcoin/rust-bitcoin/pull/4164#issuecomment-2704890639 this PR is ready to merge.

Reply:
Kix opened https://github.com/rust-bitcoin/rust-bitcoin/issues/4215

I think he's right but I don't want to hold up this PR. Let's deal with it in a followup.